k4FWCore/components/Writer.cpp

```diff
@@ -121,8 +122,8 @@ class Writer final : public Gaudi::Functional::Consumer<void(const EventContext&
     }
     iosvc->getWriter()->writeFrame(config_metadata_frame, "configuration_metadata");
 
-    if (m_metadataSvc->m_frame) {
-      iosvc->getWriter()->writeFrame(*std::move(m_metadataSvc->m_frame), podio::Category::Metadata);
+    if (const auto* metadata_frame = std::as_const(*m_metadataSvc).getFrame(); metadata_frame) {
```

Comment:
Does it need to be a const pointer?

Reply:
Do you mean `std::as_const`? Then yes, it helps resolve the const overload conflict with access modifiers.
There is a public const method and protected non-const. The constness is resolved first so a compiler tries to use the protected overload but it's not accessible and errors